registry/zookeeper/listener.go

```diff
@@ -37,18 +37,26 @@ import (
 
 // RegistryDataListener ...
 type RegistryDataListener struct {
-	interestedURL []*common.URL
-	listener      config_center.ConfigurationListener
+	subscribed map[*common.URL]config_center.ConfigurationListener
```

Comment:
@CodingSinger pls do not use pointer as map key.

Reply:
why？ I see the golang spec includes a pointer-keyed map as an example.

Reply:
> why？ I see the golang spec includes a pointer-keyed map as an example.

show me the link.

Reply:
> why？ I see the golang spec includes a pointer-keyed map as an example.

ok, you win. by https://golang.org/ref/spec#Map_types.

Reply:
win-win